src/cli.ts

```diff
@@ -2,190 +2,148 @@
 import fs from 'node:fs';
 import { mkdir, readFile, writeFile } from 'node:fs/promises';
 import path from 'node:path';
-import { parseArgs } from 'node:util';
+import { Command } from 'commander';
 import envPaths from 'env-paths';
 import * as api from './api.js';
 import { discover } from './discovery.js';
 
+const program = new Command();
 const configDir = envPaths('twinklyjs').config;
 const configFile = path.join(configDir, 'config.json');
-
 interface Config {
 	ip: string;
 }
 
 async function getConfig(): Promise<Config | null> {
 	try {
 		const fileContents = await readFile(configFile, 'utf-8');
-		const config = JSON.parse(fileContents);
-		return config;
-	} catch (err) {
+		return JSON.parse(fileContents);
+	} catch {
 		return null;
 	}
 }
 
-const { values, positionals } = parseArgs({
-	options: {
-		ip: { type: 'string' },
-		help: { type: 'boolean' },
-	},
-	tokens: true,
-	allowPositionals: true,
-});
-
-if (positionals.length === 0 || values.help) {
-	showHelp();
-	process.exit(1);
+async function saveConfig(ip: string) {
+	if (!fs.existsSync(configDir)) {
+		await mkdir(configDir, { recursive: true });
+	}
+	await writeFile(configFile, JSON.stringify({ ip }), 'utf-8');
 }
 
-const command = positionals[0];
+async function requireIP(ip: string | undefined): Promise<string> {
+	if (ip) return ip;
 
-const { ip } = values;
-function requireIP() {
-	if (!ip) {
-		showHelp();
-		process.exit(1);
-	}
-	api.init(ip);
+	const config = await getConfig();
+	if (config?.ip) return config.ip;
+
+	console.error('Error: No IP address provided or saved in the config.');
+	process.exit(1);
 }
 
-const config = await getConfig();
-if (!config) {
-	if (!['discover', 'config'].includes(command)) {
-		requireIP();
-	}
-} else {
-	if (ip) {
+program
+	.name('twinkly')
+	.description('CLI tool for managing Twinkly smart lights')
+	.version('1.0.0');
+
+program
+	.command('discover')
+	.description('Discover Twinkly devices on the network')
+	.action(async () => {
+		const devices = await discover();
+		console.log(devices);
+	});
+
+program
+	.command('getmovie')
+	.description('Get the current movie')
+	.option('--ip <ip>', 'The IP address of the Twinkly device')
```

Comment:
You don't have to do this for this commit - but we should look to see if there's a more generic way we can accept `ip` for most of the commands.